### PR TITLE
Add Thruk as Grafana Datasource

### PR DESCRIPTION
**Pull request by rverchere**
##### SUMMARY

Add Thruk as a Grafana Datasource

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Updates the grafana_datasource module

##### ADDITIONAL INFORMATION
Replaces the following commands:

```bash
grafana-cli plugins install sni-thruk-datasource
```

This will install this datasource plugin: https://grafana.com/plugins/sni-thruk-datasource

### CODE REVIEW

**Comment by seuf** [comment]
We plan to refactor grafana_datasource module to have a common "utils" library, and smaller python scripts for each datasource. If we merge it now @rrey you'll need to rebase your work in https://github.com/ansible-collections/grafana/pull/18 to add this datasource in a separated .py like the other.

**Comment by rverchere** [comment]
> We plan to refactor grafana_datasource module to have a common "utils" library, and smaller python scripts for each datasource. If we merge it now @rrey you'll need to rebase your work in #18 to add this datasource in a separated .py like the other.

I can create a new PR once it's done if you want, or help in any other way ;)

**Comment by gundalow** [comment]
@rverchere Thank you for migrating your patch to this new repo, and for beta testing this new process for us.

Was it obvious what was needed to be done? 
Was there anything that we would explain better?

**Comment by rverchere** [comment]
> @rverchere Thank you for migrating your patch to this new repo, and for beta testing this new process for us.

@gundalow no pb!
 
> Was it obvious what was needed to be done?

It was easy, my change is not a big deal, even if the `git diff`+ `patch` did not work as files have been renamed.

> Was there anything that we would explain better?

Yes please, How can I add integration tests (any pointers to documentation is welcome), as suggested by @rrey ? How can I test it locally (or on my own repo) ?
I guess you need something like https://github.com/ansible-collections/grafana/runs/359891138#step:5:804 ?

**Comment by rrey** [comment]
I think you can read the official ansible documentation first: [here](https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#integration-tests)
Things will be easier for collections as there is no env-setup required.

Then, here is a QuickStart:

1. Use ansible 2.9.x
2. start Grafana locally using docker, and make it accessible on 127.0.0.1:3000
3. clone this repository in the following directory tree : `ansible_collections/community/`.
The tree is important to have `ansible-test` command working.
4. once you have clone the repo, you can check that things are working by running `ansible-test`:
```
$ ansible-test integration --docker 
``` 
It will trigger the integration tests. On master it should work without errors.

On your side, you need to add a test for the `grafana_datasource` module, so you'll have to update the role located in `tests/integration/targets/grafana_datasource/`. The tests are splitter by datasource type, so you just need to add a file with the module calls and the proper asserts.
You can check the other datasources tests for examples.

Shout out if you need help.

File: tests/integration/targets/grafana_datasource/tasks/thruk.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    name: datasource-thruk
+    grafana_url: "{{ grafana_url }}"
+    grafana_user: "{{ grafana_username }}"
+    grafana_password: "{{ grafana_password }}"
+    org_id: "1"
+    ds_type: sni-thruk-datasource
+    url: "https://thruk.company.com/sitename/thruk"
+    tls_skip_verify : yes
+    validate_certs: no
+- debug:
```
Comment (rrey) [inline tests/integration/targets/grafana_datasource/tasks/thruk.yml]: please remove the debug task here.

Comment (rrey) [reply tests/integration/targets/grafana_datasource/tasks/thruk.yml]: Please also add an empty line between tasks to ease readability.

Comment (rverchere) [reply tests/integration/targets/grafana_datasource/tasks/thruk.yml]: > please remove the debug task here.

Should I remove also the debug task in idempotency one?

Comment (rrey) [reply tests/integration/targets/grafana_datasource/tasks/thruk.yml]: yep, the debug tasks should not remain in the tests roles.

**Comment by gundalow** [comment]
@all-contributors add @rverchere for code

**Pull request merged**